src/Service/Injector.php

```diff
@@ -42,7 +43,28 @@ public function __construct(ContainerFactory $containerFactory, ExtractorFactory
      */
     public function inject(/*object */$object): void
     {
-        \array_map($this->getPropertyInjector($object), $this->extractProperties($object));
+        $props = $this->extractProperties($object);
+        $visitedProps = [];
+        foreach ($props as $index => $prop) {
+            $key = $prop->getPropertyName();
+            if (!isset($visitedProps[$key])) {
+                $visitedProps[$key] = $prop;
+
+                continue;
+            }
+
+            if ($prop->privatized()) {
```

Comment:
There is still one case which is not supported I believe: https://3v4l.org/hB6dI

Reply:
The child property is not private but the parent is => It should be possible to inject different services in both properties.